src/graph/gremlinEndpoints.ts

```diff
@@ -4,23 +4,12 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { CosmosDBManagementClient } from '@azure/arm-cosmosdb';
-import { nonNullProp, nonNullValue } from '../utils/nonNull';
+import { nonNullValue } from '../utils/nonNull';
 import { IGremlinEndpoint } from '../vscode-cosmosdbgraph.api';
 
 export async function tryGetGremlinEndpointFromAzure(client: CosmosDBManagementClient, resourceGroup: string, account: string): Promise<IGremlinEndpoint | undefined> {
-    const response = await new Promise((resolve, reject) => {
-        // Use the callback version of get because the Promise one currently doesn't expose gremlinEndpoint (https://github.com/Azure/azure-documentdb-node/issues/227)
-        client.databaseAccounts.get(resourceGroup, account, (error, _result, _httpRequest, innerResponse) => {
-            if (error) {
-                reject(error);
-            } else {
-                resolve(innerResponse);
-            }
-        });
-    });
-
-    const body: string = nonNullProp(<{ body?: string }>response, 'body');
-    const endpointUri = JSON.parse(body).properties?.gremlinEndpoint;
+    const response: string = (await client.databaseAccounts.get(resourceGroup, account))._response.bodyAsText;
```

Comment:
Just like before, this should have a comment explaining why you're doing something weird